astropy/io/fits/diff.py

```diff
@@ -363,20 +357,10 @@ def _diff(self):
             a_names = [hdu.name for hdu in self.a]
             b_names = [hdu.name for hdu in self.b]
             for pattern in self.ignore_hdu_patterns:
-                self.a = HDUList(
-                    [
-                        h
-                        for h in self.a
-                        if h.name not in fnmatch.filter(a_names, pattern)
-                    ]
-                )
-                self.b = HDUList(
-                    [
-                        h
-                        for h in self.b
-                        if h.name not in fnmatch.filter(b_names, pattern)
-                    ]
-                )
+                a_filt = fnmatch.filter(a_names, pattern)
```

Comment:
Bit sad that `black` forces one to do these work-arounds to keep things tight. If it fits on the line, I'd call these `a_ignore` or `a_exclude` instead of `a_filt` (which to me suggests things that are included, not excluded). But no big deal.